client/src/model/component/Link.hx

```diff
@@ -147,7 +147,7 @@ class Link extends CircuitElement implements Observer {
     }
 
     //the distance of point (x0,y0) to line [(x1,y1),(x2,y2)]
-    function pointToLine(x1:Float, y1:Float, x2:Float, y2:Float, x0:Float, y0:Float):Float{
+    public static function pointToLine(x1:Float, y1:Float, x2:Float, y2:Float, x0:Float, y0:Float):Float{
```

Comment:
This method and the next can certainly be static.  Maybe they should be moved out of Line to some utility class for doing geometric calculations.